### PR TITLE
[KOGITO-5796] BDD tests: Adjust Knative test to use KogitoSource for …

### PR DESCRIPTION
**Pull request by sutaakar**
…events

https://issues.redhat.com/browse/KOGITO-5796

Signed-off-by: Karel Suta <ksuta@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: test/pkg/steps/knative-eventing-kogito.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	addDefaultJavaOptionsIfNotProvided(kogitoRuntime.GetSpec())
+
+	// TODO: Quick workaround, needs to be refactored once KogitoSource API is moved to separate module
+	kogitoSourceContent := `apiVersion: kogito.knative.dev/v1alpha1
+kind: KogitoSource
+metadata:
+  name: process-knative-quickstart-quarkus
+spec:
+  image: quay.io/kiegroup/examples-process-knative-quickstart-quarkus-nightly:latest
```
Comment (radtriste) [inline test/pkg/steps/knative-eventing-kogito.go:46]: Could we make that image more flexible with a replace ?
Else the release pipeline will not test with the soon-to-be-released image

File: test/pkg/steps/knative-eventing-kogito.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	ctx.Step(`^Create (quarkus|springboot) KogitoSource "([^"]*)" sinking events to Broker "([^"]*)" from runtime registry$`, data.createKogitoSource)
+}
+
+func (data *Data) installKnativeEventingKogitoSource() error {
+	return installers.GetKnativeEventingKogitoInstaller().Install(data.Namespace)
+}
+
+func (data *Data) createKogitoSource(runtimeType, kogitoSourceName, brokerName string) error {
+	imageTag := data.ScenarioContext[GetBuiltRuntimeImageTagContextKey(kogitoSourceName)]
+	kogitoRuntime := framework.GetKogitoRuntimeStub(data.Namespace, runtimeType, kogitoSourceName, imageTag)
```
Comment (radtriste) [inline test/pkg/steps/knative-eventing-kogito.go:36]: what is the kogitoRuntime for ?

Comment (sutaakar) [reply test/pkg/steps/knative-eventing-kogito.go:36]: good catch, originally I was planning to to make creation of KogitoSource dynamic using programmatic API. But it drags many unneeded dependencies so I took a hardcoded approach as a temporary workaround. Will remove the kogitoRuntime.

File: test/pkg/steps/knative-eventing-kogito.go (hunk 1 above)
Comment (sutaakar) [reply test/pkg/steps/knative-eventing-kogito.go]: Done

**Review by radtriste — APPROVED** [review]

**Comment by sutaakar** [comment]
Test failures unrelated

**Pull request merged**